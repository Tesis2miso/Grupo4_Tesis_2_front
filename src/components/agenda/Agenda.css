.login-page {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100vh;
  }
  
  .login-header {
    display: flex;
    justify-content: flex-start;
    margin: 2rem;
    color: rgb(24, 24, 24);
  }
  
  .login-title {
    margin-left: 2rem;
    font-size: 2rem;
    font-weight: bold;
  }
  
  .agenda {
    width: 900px;
    margin: 0 auto;
    padding: 20px;
    border-radius: 10px;
    box-shadow: 0 2px 6px rgba(0,0,0,0.1);
    background-color: #fff;
  }
  
  .agenda-header {
    background-color: #3b5998;
    border-bottom: 1px solid #bbb;
    padding: 10px;
    color: white
  }
  
  .agenda-body {
    margin-top: 10px;
  }
  
  .agenda-row {
    display: flex;
    margin-bottom: 10px;
  }
  
  .agenda-time {
    background-color: #ddd;
    border: 1px solid #bbb;
    border-radius: 5px;
    color: #555;
    font-weight: bold;
    padding: 10px;
    width: 100px;
    text-align: center;
  }
  
  .agenda-event {
    background-color: #fff;
    border: 1px solid #bbb;
    border-radius: 5px;
    padding: 10px;
    margin-left: 10px;
    flex-grow: 1;
  }

  .agenda-patient {
    background-color: #fff;
    border: 1px solid #bbb;
    border-radius: 5px;
    padding: 10px;
    margin-left: 10px;
    flex-grow: 1;
  }
 
  .agenda-table {
    margin: 0 auto;
    background-color: rgb(123, 29, 29);
    border-radius: 10px;
    overflow: hidden;
    margin-bottom: 50px;
    width: 900px;
  }

  th,
  td {
    padding: 15px;
    background-color: rgb(233, 229, 229); 
  
  }

  th {
    background-color: #4f4f4f;
    color: #fff;
    font-weight: bold;
  }

  tr:hover {
    background-color: rgb(122, 77, 77);
  }

  td:hover {
    background-color: rgb(249, 227, 32);
  }