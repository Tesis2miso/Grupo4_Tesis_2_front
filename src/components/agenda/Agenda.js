import { useState, Fragment } from "react";
import axios from "axios";
import "./Agenda.css";
import { useTranslation } from "react-i18next";
import { useEffect } from "react";
import i18n from '../utils/i18n'

function Agenda(props) {
  const [locale, setLocale] = useState(i18n.language);
  i18n.on('languageChanged', (lng) => setLocale(i18n.language));
  const { t } = useTranslation();

  const [scheduleData, setScheduleData] = useState([]);

  useEffect(() => {
    async function fetchData() {
      axios({
        method: "GET",
        headers: {
          Authorization: "Bearer " + localStorage.getItem("token"),
        },
        url: `${process.env.REACT_APP_BASE_PATH}/agenda_specialist`,
      })
        .then((response) => {
          if (response) {
            setScheduleData(response.data);
          }
        })
        .catch((error) => {
          //console.log(error);
        });
    }
  
    fetchData();
  }, []);

  return (
    <Fragment>
      <div></div>
      <div className="login-page">
        <div className="login-header"></div>
        <div>
          <div className="agenda-body">
            <table id="agenda-table">
              <thead>
                <tr>
                  <th className="agenda-header">{t("Time")}</th>
                  <th className="agenda-header">{t("Detail")}</th>
                  <th className="agenda-header">{t("Patient")}</th>
                  <th className="agenda-header">{t("Patient email")}</th>
                  <th className="agenda-header">{t("Date of appointment")}</th>
                </tr>
              </thead>
              <tbody>
                {scheduleData.map((item, index) => (
                  <tr key={index}>
                    <td>{item.created_at.substring(11, 16)}</td>
                    <td>
                      {item.injury_type} 
                    </td>
                    <td>{item.user_name}</td>
                    <td>{item.user_email}</td>
                    <td>{item.created_at.substring(0, 10)}</td>
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </Fragment>
  );
}

export default Agenda;
