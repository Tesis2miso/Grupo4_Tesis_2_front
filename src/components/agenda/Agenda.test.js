import '@testing-library/jest-dom';
import { mount, shallow } from 'enzyme';
import Agenda from './Agenda';
import axios from "axios";
import { act } from 'react-dom/test-utils';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
jest.mock('axios', () => jest.fn());
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => { }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => { },
  }
}));
describe('Tests_Agenda', () => {
  let wrapper = null;
  let consults = [];

  beforeEach(() => {
    wrapper = shallow(<Agenda loggedIn={true} setToken={() => { }} />);
    consults = [ {
        created_at: '2022-02-19T10:00:00.000Z',
        injury_type: 'Acne',
        shape: 'Redonda',
        injuries_count: 3,
        user_name: 'Jane Doe',
        user_email: 'jane@example.com'
      }]
  });

  test('show agenda', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('load agenda', async () => {
    axios.mockResolvedValueOnce(Promise.resolve({ data: consults }));
    const prop = {
      OnValChange: jest.fn()
    }
    await act(async () => {
      mount(<Agenda {...prop} />)
    });
  })

  test('load agenda with error', async () => {
    axios.mockResolvedValueOnce(Promise.reject({ response: { data: { mssg: '' } } }));
    const prop = {
      OnValChange: jest.fn()
    }
    await act(async () => mount(<Agenda {...prop} />));
  })
})