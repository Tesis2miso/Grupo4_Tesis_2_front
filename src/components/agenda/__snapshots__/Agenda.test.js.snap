// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests_Agenda show agenda 1`] = `
<Fragment>
  <div />
  <div
    className="login-page"
  >
    <div
      className="login-header"
    />
    <div>
      <div
        className="agenda-body"
      >
        <table
          id="agenda-table"
        >
          <thead>
            <tr>
              <th
                className="agenda-header"
              >
                Hora
              </th>
              <th
                className="agenda-header"
              >
                Detalle
              </th>
              <th
                className="agenda-header"
              >
                Paciente
              </th>
              <th
                className="agenda-header"
              >
                Email paciente
              </th>
              <th
                className="agenda-header"
              >
                Fecha
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</Fragment>
`;
